dialog/fileicon.go

```diff
@@ -97,8 +96,7 @@ func (s fileIconRenderer) Refresh() {
 	canvas.Refresh(s.item)
 }
 
-func mimeTypeGet(path string) (ext, mimeType, mimeSubType string) {
-	uri := storage.NewURI(path)
+func mimeTypeGet(uri fyne.URI) (ext, mimeType, mimeSubType string) {
```

Comment:
The name of this function confuses me.
AFAICS there is one usage for all delivered details and one new use which only needs the mime type information.
Because the extension and the mime type are two different things (one is a property of the file path the other one is a property of the file data (strictly speaking)), I suggest to split it into two methods like `extensionForURI` and `mimeTypeForURI`.

Reply:
extensionForURI would just be uri.Extension() ;)

Reply:
yes both already exist uri.Mimetype and uri.Extension ... This function returns the uri.MimeType split into type and subtype for use in creating file icons and filtering on dialogs.  It returns the extension too because that was needed to build out the File Icon and it would involve one less call to uri.Extension since we were already calling it in that funciton. I'll remove extension from the return though and just call uri.Extension again.